src/config.rs

```diff
@@ -111,6 +114,13 @@ fn validate_maintainer(maintainer: &String) -> Result<(), String> {
     Err("Invalid format.".to_owned())
 }
 
+fn validate_arch(arch: &String) -> Result<(), String> {
+    match check_arch_name(arch) {
```

Comment:
Use `check_arch_name(arch).context("Invalid architecture name")`

Reply:
This is written as a validator for dialoger, which requires `Result<(), String>`.

Reply:
Oh year sorry, I forgot to change the function signature.